src/topology/algebra/ordered.lean

```diff
@@ -2348,3 +2404,65 @@ lemma continuous_at_iff_continuous_left'_right' [topological_space α] [linear_o
   continuous_at f a ↔ continuous_within_at f (Iio a) a ∧ continuous_within_at f (Ioi a) a :=
 by rw [continuous_within_at_Ioi_iff_Ici, continuous_within_at_Iio_iff_Iic,
   continuous_at_iff_continuous_left_right]
+
+section homeomorphisms
+variables [topological_space α] [topological_space β]
+
+section linear_order
+variables [linear_order α] [order_topology α]
+variables [linear_order β] [order_topology β]
+
+/-- If `f : α → β` is strictly monotone and surjective, it is everywhere continuous. -/
+lemma continuous_at_of_strict_mono_surjective
+  {f : α → β} (h_mono : strict_mono f) (h_surj : function.surjective f) (a : α) :
+  continuous_at f a :=
+continuous_at_iff_continuous_left_right.mpr
+  ⟨continuous_left_of_strict_mono_surjective h_mono h_surj a,
+  continuous_right_of_strict_mono_surjective h_mono h_surj a⟩
+
+/-- If `f : α → β` is strictly monotone and surjective, it is continuous. -/
+lemma continuous_of_strict_mono_surjective
+  {f : α → β} (h_mono : strict_mono f) (h_surj : function.surjective f) :
+  continuous f :=
+continuous_iff_continuous_at.mpr (continuous_at_of_strict_mono_surjective h_mono h_surj)
+
+/-- If `f : α ≃ β` is strictly monotone, its inverse is continuous. -/
+lemma continuous_inv_of_strict_mono_equiv (e : α ≃ β) (h_mono : strict_mono e.to_fun) :
+  continuous e.inv_fun :=
+begin
+  have hinv_mono : strict_mono e.inv_fun,
+  { intros x y hxy,
+    rw [← h_mono.lt_iff_lt, e.right_inv, e.right_inv],
+    exact hxy },
+  have hinv_surj : function.surjective e.inv_fun,
+  { intros x,
+    exact ⟨e.to_fun x, e.left_inv x⟩ },
+  exact continuous_of_strict_mono_surjective hinv_mono hinv_surj
+end
+
+/-- If `f : α → β` is strictly monotone and surjective, it is a homeomorphism. -/
+noncomputable def homeomorph_of_strict_mono_surjective
+  {f : α → β} (h_mono : strict_mono f) (h_surj : function.surjective f) :
+  homeomorph α β :=
+{ to_equiv := equiv.of_bijective f ⟨strict_mono.injective h_mono, h_surj⟩,
+  continuous_to_fun := continuous_of_strict_mono_surjective h_mono h_surj,
+  continuous_inv_fun := continuous_inv_of_strict_mono_equiv
+    (equiv.of_bijective f ⟨strict_mono.injective h_mono, h_surj⟩) h_mono }
```

Comment:
Please make the argument `f` explicit, otherwise pretty printer will show `homeomorph_of_strict_mono_surjective _ _` whenever proofs are nontrivial.

Reply:
And `def` should go with `@[simp] lemma coe_homeomorph_of_strict_mono_surjective : \u=(homeomorph... ) = f := rfl`